modules/stereo/src/descriptor.hpp

```diff
@@ -0,0 +1,441 @@
+//any express or implied warranties, including, but not limited to, the implied
+//warranties of merchantability and fitness for a particular purpose are disclaimed.
+//In no event shall copyright holders or contributors be liable for any direct,
+//indirect, incidental, special, exemplary, or consequential damages
+//(including, but not limited to, procurement of substitute goods or services;
+//loss of use, data, or profits; or business interruption) however caused
+//and on any theory of liability, whether in contract, strict liability,
+//or tort (including negligence or otherwise) arising in any way out of
+//the use of this software, even if advised of the possibility of such damage.
+
+/*****************************************************************************************************************\
+*   The interface contains the main descriptors that will be implemented in the descriptor class                  *
+\******************************************************************************************************************/
+
+#include "precomp.hpp"
+#include <stdint.h>
+#ifndef _OPENCV_DESCRIPTOR_HPP_
+#define _OPENCV_DESCRIPTOR_HPP_
+#ifdef __cplusplus
+
+namespace cv
+{
+    namespace stereo
+    {
+        enum { Dense_Census, Sparse_Census, StarCensus};
+        enum {ClassicCenterSymetricCensus, ModifiedCenterSymetricCensus};
+        enum {StandardMct,MeanVariation};
+        enum {SSE, NonSSE};
+        //!Mean Variation is a robust kernel that compares a pixel
+        //!not just with the center but also with the mean of the window
+        struct MVKernel
+        {
+            uint8_t *image1;
+            uint8_t *image2;
+            uint8_t *integralLeft;
+            uint8_t *integralRight;
+            MVKernel(uint8_t *img, uint8_t *img2, uint8_t *integralL, uint8_t *integralR): image1(img),image2(img2),integralLeft(integralL), integralRight(integralR){}
+            void operator()(int rrWidth,int w2, int rWidth, int jj, int j, int &c, int &c2) const
+            {
+                (void)rrWidth;
+                (void)w2;
+                (void)rWidth;
+                (void)jj;
+                (void)j;
+                (void)c;
+                (void)c2;
+            }
+        };
+        //single version of MV
+        struct MVKernelSingle
+        {
+            uint8_t *image1;
+            uint8_t *integralLeft;
+            MVKernelSingle(uint8_t *img, uint8_t *integralL): image1(img),integralLeft(integralL){}
+            void operator()(int rrWidth,int w2, int rWidth, int jj, int j, int &c) const
+            {
+                (void)rrWidth;
+                (void)w2;
+                (void)rWidth;
+                (void)jj;
+                (void)j;
+                (void)c;
+            }
+        };
+        //!kernel that takes the pixels from certain positions from a patch
+        //!offers verry good results
+        struct StarKernel
```

Comment:
Please, remove all the "yet to be implemented" functions. There's no point in adding them without the implementation.